src/couch_replicator_scheduler.erl

```diff
@@ -381,9 +387,19 @@ pending_job_count() ->
     ets:select_count(?MODULE, [{#job{pid=undefined, _='_'}, [], [true]}]).
 
 
--spec pending_jobs() -> [#job{}].
-pending_jobs() ->
-    ets:match_object(?MODULE, #job{pid=undefined, _='_'}).
+% Get list of not recently crashed job infos {#job.id, last_started(Job)}.
+% A smaller tuple is used here to reduce memory usage due to returning
+% potentially a huge list of pending job records.
+-spec pending_jobs_info() -> [{job_id(), erlang:timestamp()}].
+pending_jobs_info() ->
```

Comment:
What do you think about passing in `Count` to `pending_jobs_info` and only accumulating the top `Count` entries? This still accumulates an element for ~every entry in the table.